sqldb/postgres.go

```diff
@@ -0,0 +1,90 @@
+package sqldb
+
+import (
+	"database/sql"
+	"fmt"
+
+	postgres_migrate "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file" // Register driver
+	"github.com/lightningnetwork/lnd/sqldb/sqlc"
+)
+
+const (
+	dsnTemplate = "postgres://%v:%v@%v:%d/%v?sslmode=%v"
+)
+
+// PostgresConfig holds the postgres database configuration.
+//
+//nolint:lll
+type PostgresConfig struct {
+	SkipMigrations     bool   `long:"skipmigrations" description:"Skip applying migrations on startup."`
+	Host               string `long:"host" description:"Database server hostname."`
+	Port               int    `long:"port" description:"Database server port."`
+	User               string `long:"user" description:"Database user."`
+	Password           string `long:"password" description:"Database user's password."`
+	DBName             string `long:"dbname" description:"Database name to use."`
+	MaxOpenConnections int32  `long:"maxconnections" description:"Max open connections to keep alive to the database server."`
+	RequireSSL         bool   `long:"requiressl" description:"Whether to require using SSL (mode: require) when connecting to the server."`
+}
+
+// DSN returns the dns to connect to the database.
+func (s *PostgresConfig) DSN(hidePassword bool) string {
+	var sslMode = "disable"
+	if s.RequireSSL {
+		sslMode = "require"
+	}
+
+	password := s.Password
+	if hidePassword {
+		// Placeholder used for logging the DSN safely.
+		password = "****"
+	}
+
+	return fmt.Sprintf(dsnTemplate, s.User, password, s.Host, s.Port,
+		s.DBName, sslMode)
+}
+
+// NewPostgresStore creates a new store that is backed by a Postgres database
+// backend.
+func NewPostgresStore(cfg *PostgresConfig) (*SQLDB, error) {
+	log.Infof("Using SQL database '%s'", cfg.DSN(true))
+
+	// TODO(positiveblue): use a pool of connections if possible.
+	db, err := sql.Open("pgx", cfg.DSN(false))
+	if err != nil {
+		return nil, err
+	}
+
+	if !cfg.SkipMigrations {
+		// Now that the database is open, populate the database with
+		// our set of schemas based on our embedded in-memory file
+		// system.
+		//
+		// First, we'll need to open up a new migration instance for
+		// our current target database: sqlite.
+		driver, err := postgres_migrate.WithInstance(
+			db, &postgres_migrate.Config{},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		postgresFS := newReplacerFS(sqlSchemas, map[string]string{
+			"BLOB":                "BYTEA",
+			"INTEGER PRIMARY KEY": "SERIAL PRIMARY KEY",
+			"TIMESTAMP":           "TIMESTAMP WITHOUT TIME ZONE",
```

Comment:
I'm not sure about the implications of storing with or without the time zone. E.g. assuming I'm running my node in my home in utc+0 and later migrating that to a remote server that is in a different timezone, my gut feeling is that would be a conflict here?

Reply:
I think it's ok to use no timezone until we make sure to display all printed timestamps as UTC or alternatively transform to the user's timezone (when printing).

Reply:
@sputn1ck your gut feeling is right here:  `time/timezone == software engineering headache`

As  Andras said, the idea is to not use timezone (and everything is sitll consistent) until you want to show it to the user